### DISCUSSION TITLE
index_reader.doc_contents(key) return None while index_reader.doc_raw(key) and index_reader.doc(key) can retrieve document

### DISCUSSION THREAD

**Discussion by zhiyuanpeng** [castorini/pyserini#1743] [Q&A] [answered]
Hi there,

After I build index on my documents with command:
```
command = f"python -m pyserini.index -collection JsonCollection \
-generator DefaultLuceneDocumentGenerator -threads {threads} \
-input {self.doc_dir} -index {self.index_dir} -storeRaw \
-storePositions -storeDocvectors"
```
I try to check the analyzed documents by:

```
from pyserini.index.lucene import IndexReader
index_reader = IndexReader(index_dir)
anserini_indexed = index_reader.doc_contents(key)
anserini_raw = index_reader.doc_raw(key)
anserini_doc = index_reader.doc(key)
```
anserini_doc and anserini_raw are not None, but anserini_indexed is None. It seems that the command I used to build the index doesn't keep the analyzed documents. I find the options for building index in this [page](https://github.com/castorini/anserini/blob/master/docs/common-indexing-options.md) do not have relevant options.

How to keep the analyzed documents when building the index? Thanks.

**Comment by lintool** [comment]
Try: https://github.com/castorini/pyserini/blob/master/docs/usage-fetch.md

**Reply by zhiyuanpeng** [reply]
Thanks. I just found it in the java source file.

**Reply by zhiyuanpeng** [reply]
I just use `-storeContents` to build the index again and find that `index_reader.doc_contents(key)` returns the unanalyzed string of the document instead of the analyzed tokens. How to find the analyzed term of each document in the index?  Otherwise, I have to call the analyzer to analyze the documents again. Thanks.

**Reply by lintool** [reply, marked as answer]
Are you sure you've read the docs? https://github.com/castorini/pyserini/blob/master/docs/usage-indexreader.md#how-do-i-access-and-manipulate-term-vectors

Is this what you are looking for?

**Reply by zhiyuanpeng** [reply]
Thanks, that is what I am looking for.

**Reply by zhiyuanpeng** [reply]
BTW, it seems that there is a typo in the doc:

```
term = 'cities'

# Analyze the term.
analyzed = index_reader.analyze(term)
print(f'The analyzed form of "{term}" is "{analyzed[0]}"')

# Skip term analysis:
df, cf = index_reader.get_term_counts(analyzed[0], analyzer=None)
print(f'term "{term}": df={df}, cf={cf}')
```
as when skip term analysis, term should be passed in instead of analyzed[0]
```
df, cf = index_reader.get_term_counts(term, analyzer=None)
```

**Reply by lintool** [reply]
Can you please send a PR to fix?
Even better, can you please add a test case?

**Reply by zhiyuanpeng** [reply]
Sure, will do it.

**Reply by zhiyuanpeng** [reply]
I find the readme file is correct actually and there is no need to fix it.
